Call check_shields when attacked by an abductor baton.

nullquery opened the pull request:
Fixes #14599.

Tested by setting myself to abductor and spawning an interactive human mob.

Without reactive armor stunning works in both cases.

Before patch: With reactive armor the mob is NOT teleported. However, a regular stun baton does initiate a teleport.

After patch: With reactive armor the mob is now teleported just as with a regular stun baton.

I've taken the assumption that probing a mob with the abductor baton counts as an aggressive action and therefore initiates teleport. From what I could see this would only be a balance issue if the person being probed is a changeling -- they would be able to escape exposure by wearing reactive teleport armor. I don't think this is a problem.

This is also my first patch using Git directly, so please let me know if there are issues with the patch.

Commit:
Call check_shields when attacked by an abductor baton.

nullquery commented on code/game/gamemodes/miniantags/abduction/abduction_gear.dm:
```diff
@@ -411,20 +411,28 @@ Congratulations! You are now trained for xenobiology research!"}
 	user.do_attack_animation(L)
-	switch(mode)
+
+	// [tgstation/#14599] Allow reactive teleport armor to kick in when attacked.
```
Note that I've added this line to point to the bug that was fixed. I did this so that if someone has any problems with this addition in the future they'll know why it was added and be informed enough to assess if/how they should change it.

I just copied it from the stun baton code to quickly address this bug, but ideally we'd want to avoid copy/pasting code like this and work toward generalizing these kinds of things. Is there an object that shouldn't trigger a mob's defenses? If there isn't, it may be possible to resolve this in the calling proc.

RemieRichards replied:
We typically don't label bug fixes in the code.

nullquery replied:
That's a bad habit!

I'm all for removing that comment _if_ the original author of the `check_shields` proc shows up and tells me that this is a good fix. Otherwise I can't guarantee that this fix hasn't introduced a new bug -- and in that case you'll want to know why someone put that code there.

nullquery commented on code/game/gamemodes/miniantags/abduction/abduction_gear.dm:
```diff
@@ -411,20 +411,28 @@ Congratulations! You are now trained for xenobiology research!"}
+	if (ishuman(L))
+		var/mob/living/carbon/human/H = L
+		if (H.check_shields(0, "[user]'s [name]", src, MELEE_ATTACK))
+			playsound(L, 'sound/weapons/Genhit.ogg', 50, 1)
```
Unsure if this line is necessary, but I copied this from the regular stun baton code so I'm not the right person to make that call.

Commit:
Remove unnecessary comment in favor of Git blame feature.

Fixes #14599

duncathan commented on code/game/gamemodes/miniantags/abduction/abduction_gear.dm:
```diff
@@ -411,20 +411,27 @@ Congratulations! You are now trained for xenobiology research!"}
 		return
 
-	if(!isliving(target))
+	if (!isliving(target))
```
https://github.com/tgstation/-tg-station/blob/master/CONTRIBUTING.md#operators-and-spaces

Just for consistency.

duncathan commented:
:+1:

Commit:
Directions have been given on how to proceed.
Underestimated the code style of this branch.
Now I've fixed it.
Clearly I was mistaken.
Forgive my mistakes.
Once upon a time...
Really, a long time ago.
Certain decisions were made.
Everyone agreed at the time.
Some didn't.
Maybe we could change the standard.
Eh, let's not.

GunHog commented:
Not sure how I feel on this one, to be honest. However, about probing: It is about as aggressive as scanning someone with a health analyzer. Being exposed as a changeling to the abductors is a good thing to the victim, as smart abductors will *back off* from a ling. They can fight back much more easily than others. I do not think that one truly warrants triggering armor or shields.

duncathan commented:
@GunHog probing is about as aggressive as hugging

Shadowlight213 commented:
Let probing bypass it. Probing is a inherent right of ayylmaos that must not be denied.

GunHog commented:
> probing is about as aggressive as hugging

I agree entirely, Dunc! Did you know that hugging bypasses the armor as well? You are therefore supporting my argument!

duncathan commented:
shit
bring back hugging kills 2k16

tkdrg commented:
Who cares, honestly? Ling is off rotation and an abductor probing a ling with reactive teleporter armor is a fantastically rare edge case either way,
